### PR TITLE
executor: Fix issue 1154

### PR DESCRIPTION
**Pull request by zimulala**
fix issue #1154

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -349,10 +349,21 @@ func (b *executorBuilder) buildSort(v *plan.Sort) Executor {
 
 func (b *executorBuilder) buildLimit(v *plan.Limit) Executor {
 	src := b.build(v.Src())
+
+	upperCount := v.Count + v.Offset
+	// If v.Offset value is 0, then v.Count the max value is math.MaxUint64,
```
Comment (shenli) [inline executor/builder.go]: then the max value of v.Count is math.MaxUint64,

**Comment by shenli** [comment]
Rest LGTM

File: executor/builder.go [hunk 2]
```diff
@@ -349,10 +349,21 @@ func (b *executorBuilder) buildSort(v *plan.Sort) Executor {
 
 func (b *executorBuilder) buildLimit(v *plan.Limit) Executor {
 	src := b.build(v.Src())
+
+	upperCount := v.Count + v.Offset
+	// If v.Offset value is 0, then the max value of v.Count is math.MaxUint64,
+	// so it doesn't need to do special treatment for upperCount.
+	if v.Offset != 0 {
+		// It means math.MaxUint64 less then (v.Count + v.Offset).
```
Comment (coocood) [inline executor/builder.go]: less than

File: executor/executor.go [hunk 3]
```diff
@@ -827,10 +827,11 @@ func (e *SelectLockExec) Close() error {
...
 type LimitExec struct {
-	Src    Executor
-	Offset uint64
-	Count  uint64
-	Idx    uint64
+	Src        Executor
+	Offset     uint64
+	Count      uint64
+	Idx        uint64
+	upperCount uint64
```
Comment (ngaut) [inline executor/executor.go]: s/upperCount/upperBound/

File: executor/builder.go [hunk 4]
```diff
@@ -349,10 +349,21 @@ func (b *executorBuilder) buildSort(v *plan.Sort) Executor {
...
+		if (math.MaxUint64-v.Count)/v.Offset == 0 {
```
Comment (coocood) [inline executor/builder.go]: how about 
```
if v.Count > math.MaxUint64 - v.Offset {
    v.Count = math.MaxUint64 - v.Offset
}
```

**Comment by coocood** [comment]
LGTM

**Pull request merged**